<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PathFinder</class>
 <widget class="QMainWindow" name="PathFinder">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>275</width>
    <height>414</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PathFinder</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>251</width>
      <height>291</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="Basic">
     <attribute name="title">
      <string>Basic</string>
     </attribute>
     <widget class="QLabel" name="algorithmLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Search algorithm:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="startX">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="goalX">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="goalLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>47</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Goal:</string>
      </property>
     </widget>
     <widget class="QLabel" name="startLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>47</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Start:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="goalY">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>140</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="selectAlgorithm">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Select...</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Bredth-First Search</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Depth-First Search</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>A* Search</string>
       </property>
      </item>
     </widget>
     <widget class="QLineEdit" name="startY">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>90</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="basicLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Enter the start and goal co-ordinates in the form (x, y) and choose the search algorithm to use.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="mapSizeLabel">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>70</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Map size:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="mapSize">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>90</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>10</string>
      </property>
     </widget>
     <widget class="QLabel" name="mapSizeWarning">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>120</y>
        <width>71</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Intermediate">
     <attribute name="title">
      <string>Obstacles</string>
     </attribute>
     <widget class="QTextBrowser" name="obstList">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>80</y>
        <width>91</width>
        <height>141</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="resetObst">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>230</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset Obstacles</string>
      </property>
     </widget>
     <widget class="QLabel" name="obstaclesLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Obstacles:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="newObstX">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="addObst">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Add</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="newObstY">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>90</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Enter the co-ordinates of any obstacles that PathFinder cannot travel through in the form (x, y).</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="obstWarning">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>101</width>
        <height>101</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Advanced">
     <attribute name="title">
      <string>Terrain</string>
     </attribute>
     <widget class="QLineEdit" name="newTerrainValue">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>140</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="resetTerrain">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>230</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset Terrain</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="newTerrainX">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="newTerrainY">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>140</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="terrainList">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>130</y>
        <width>91</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="terrainLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>47</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Terrain:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="addTerrain">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Add</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>211</width>
        <height>91</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter the co-ordinates and weight of any terrain changes you would like to make.&lt;/p&gt;&lt;p&gt;The higher the weight, the greater the cost of moving through that co-ordinate is and the less likely PathFinder will be to choose any path invovling that co-ordinate. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="terrainWarning">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>200</y>
        <width>111</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Presets">
     <attribute name="title">
      <string>Presets</string>
     </attribute>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>50</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Preset 1: some obstacles, passable terrain.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>100</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Preset 2: no obstacles, varying terrain.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>150</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Preset 3: some obstacles, varying terrain.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>200</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Preset 4: many obstacles, difficult terrain.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QRadioButton" name="preset1">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QRadioButton" name="preset2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>100</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QRadioButton" name="preset3">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>150</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QRadioButton" name="preset4">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>200</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QRadioButton" name="preset0">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string> No preset</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="warningLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>340</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="find">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>231</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Find!</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
